### ISSUE TITLE
Bestellungen über API nicht abrufbar?

### ISSUE THREAD

**Issue by alebasenau** [hellohq-io/hellohq.docs#145]
Hallo,

wir möchten erstellte Bestellungen, inklusive der Positionen, über die API holen und weiterverarbeiten. Leider komme ich nur über /v1/Projects(id) auf die Bestellungen, allerdings nicht auf die Positionen oder sonstige Informationen.

Ist es nicht möglich die Informationen (Positionen, etc.) für Bestellungen aus der API zu ziehen?

Danke vorab und grüße!

**Comment by monderino** [comment]
Hey @alebasenau,

leider weiß ich nicht ganz genau was du mit Bestellungen meinst. 
Meinst du den Belegtyp Bestellung?

Tendenziell kannst du verknüpfte Entitäten über ein 'Expand' abrufen. Eine Beschreibung dazu findest du unter https://developer.hellohq.io/
Ein Beispiel um in einem Project die Projektmitglieder abzurufen wäre also:
/v1/Projects(id)?$expand=Members

Solltest du mit Bestellung einen Beleg meinen, so kannst du die nicht über die Projects Endpoints abrufen.
Eine Rechnung bekommst du zum Beispiel über: /v1/Invoices
Für Bestellungen haben wir aktuell keine Endpoints.

Ich hoffe ich konnte dir etwas weiterhelfen. 
Füg sonst gerne noch weitere Informationen zu deinem Vorhaben an, z.B.: Wie dein Request und dein Repsonse aussieht.


Viele Grüße
Sven

**Comment by alebasenau** [comment]
Hallo Sven,

danke für die Antwort. Dann werden wir das über den Belegtyp "Lieferschein" lösen.

**Dazu hätte ich noch eine Frage:**
Ich erhalte beim selektieren mit /v1/DeliveryNotes sämtliche Lieferscheine, übergreifend über alle Subsysteme.
Kann ich mit $expand und/oder $filter die Query so erweitern, dass ich nur Lieferscheine mit einem bestimmten Status, und/oder einer bestimmten DocumentTemplateId und/oder auf das entsprechende Subsystem erhalte?

Ich hatte schon versucht mir eine Query zusammen zu basteln, blicke da aber leider nicht mehr durch. :(

Könntest du hier Licht ins Dunkle bringen?

Danke!

Gruß
Alejandro

**Comment by monderino** [comment]
Hey Alejandro,

ja alles klar. Nach Subsystem geht leider nicht, aber nach Template oder Status auf jeden Fall.
Ein Beispiel wäre:/v1/DeliveryNotes?$expand=Positions&$filter=DocumentTemplateId eq 111
Damit bekommst du alle Lieferscheine mit Positionen und dem Template mit der Id 111.

Viele Grüße
Sven

**Comment by alebasenau** [comment]
Hi Sven,

vielen Dank, das ist sehr hilfreich!

Kann man an die Query /v1/DeliveryNotes?$expand=Positions&$filter=DocumentTemplateId eq 4238 einen weiteren Filter anhängen, welcher zusätzlich auf den Status prüft?

Ich denke aber das dies nicht möglich ist?

Gruß
Alejandro

**Comment by monderino** [comment]
Klar. Du findest Beispiel in der Doku (https://developer.hellohq.io/).
Zum Beispiel: /v1/Invoices?$filter=InvoiceDate ge 2018-03-01 and InvoiceDate lt 2018-04-01

**Comment by alebasenau** [comment]
Vielen Dank, jetzt habe ich alles nötige und es funktioniert auch ;)
